### ISSUE TITLE
Backup Issue in 1.2.2

### ISSUE THREAD

**Issue by EmpsThomy** [influxdata/influxdb#8342]
I want to backup all data from today in order to create incremental backups on a separate server. I don't seem to be able to run a single backup command. My influxdb version is 1.2.2 and I try to execute the following shell command:
```
influxd backup -since "2017-05-01T00:00:00Z" -database "db_1013" "./temp"
```

What I get is the following:
```
 influxd backup -since "2017-05-01T00:00:00Z" -database "db_1013" "./temp"
2017/05/01 11:22:39 backing up db=db_1013 since 2017-05-01 00:00:00 +0000 UTC
2017/05/01 11:22:39 backing up metastore to temp/meta.04
2017/05/01 11:22:39 backing up db=db_1013 rp=autogen shard=79 to temp/db_1013.autogen.00079.00 since 2017-05-01 00:00:00 +0000 UTC
2017/05/01 11:22:39 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (0)...
2017/05/01 11:22:41 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (1)...
2017/05/01 11:22:42 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (2)...
2017/05/01 11:22:43 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (3)...
2017/05/01 11:22:44 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (4)...
2017/05/01 11:22:45 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (5)...
2017/05/01 11:22:46 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (6)...
2017/05/01 11:22:47 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (7)...
2017/05/01 11:22:48 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (8)...
2017/05/01 11:22:49 Download shard 79 failed copy backup to file: err=<nil>, n=0.  Retrying (9)...
2017/05/01 11:22:50 backup failed: copy backup to file: err=<nil>, n=0
backup: copy backup to file: err=<nil>, n=0
```
The database was cleanly installed and runs on a single server. I've already tried running the command as root and the temp folder has appropriate rights, because files in there are correctly being created.

I'm not sure if I am doing this wrong, or if it's just faulty behavior? I thought I'd open an issue for it here anyway.

Thanks,
Thomy

**Comment by hpbieker** [comment]
I have the same problem with the nightly build from yesterday. It looks like this is related to commit 
https://github.com/influxdata/influxdb/commit/eb4d311c0a7d142641e535f0956d39bb8fc50cac . I hope it can be fixed soon as we are not able to take backup...

![image](https://cloud.githubusercontent.com/assets/24992762/25774439/e1e6cce8-328e-11e7-9fb9-dc53a6064947.png)

**Comment by peebles** [comment]
I too am seeing this.  In my case, the database is partially backed up ... many shards are successfully written, but then right in the middle it begins to fail as described above.  In the influxdb log I see

```bash
(...)
[I] 2017-05-10T03:00:10Z Snapshot for path /var/lib/influxdb/data/db/autogen/38 written in 1.216044816s engine=tsm1
[I] 2017-05-10T03:00:15Z Snapshot for path /var/lib/influxdb/data/db/autogen/45 written in 5.371145294s engine=tsm1
[I] 2017-05-10T03:00:25Z Snapshot for path /var/lib/influxdb/data/db/autogen/31 written in 10.000973346s engine=tsm1
[I] 2017-05-10T03:00:25Z error writing snapshot from compactor: snapshots disabled engine=tsm1
[I] 2017-05-10T03:00:25Z Snapshot for path /var/lib/influxdb/data/db/autogen/22 written in 13.474332ms engine=tsm1
[I] 2017-05-10T03:00:25Z snapshots disabled service=snapshot
[I] 2017-05-10T03:00:26Z error writing snapshot from compactor: snapshots disabled engine=tsm1
[I] 2017-05-10T03:00:26Z Snapshot for path /var/lib/influxdb/data/db/autogen/22 written in 8.74735ms engine=tsm1
[I] 2017-05-10T03:00:26Z snapshots disabled service=snapshot
(...)
```

Once this happens, any attempted backup fails on the same shard it failed on originally, immediately, without trying any of the previous shards.

I am running a build I did on the master branch.  I've tried both alpine and ubuntu builds with same results.

Sees like the c,mu.RLock() mechanism is not quite working in tsdb/engine/tsm1/compact.go.

**Comment by jwilder** [comment]
@peebles #8378 should fix that issue on master.

**Comment by peebles** [comment]
I appears to have fixed the problem.  Awesome!!

**Comment by jwilder** [comment]
Fixed via #8378
